controllers/cassandrabackup_controller.go

```diff
@@ -80,20 +80,41 @@ func (r *CassandraBackupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		}
 	}
 
+	// If there is anything in progress, simply requeue the request
+	if len(backup.Status.InProgress) > 0 {
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
 	if backup.GetDeletionTimestamp() != nil {
 		if ctrlutil.ContainsFinalizer(backup, finalizerName) {
-			// Backup is being deleted, call Medusa to remove it also from storage
-			err = r.removeBackup(ctx, backup)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
-			}
+			// Deletion process has already been started, return later
+			patch := client.MergeFrom(backup.DeepCopy())
+			backup.Status.InProgress = append(backup.Status.InProgress, "removeBackup")
 
-			// Delete the finalizer to allow deletion process to finish
-			ctrlutil.RemoveFinalizer(backup, finalizerName)
-			err = r.Update(ctx, backup)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+			if err := r.Status().Patch(context.Background(), backup, patch); err != nil {
+				r.Log.Error(err, "failed to patch status with backup deletion start time", "Backup", backup)
```

Comment:
This log message is confusing. I don't see where we are setting the deletion start time in the status.